src/Compilers/CSharp/Portable/CommandLine/CommandLineParser.cs

```diff
@@ -1168,6 +1191,16 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string bas
             };
         }
 
+        private string TrimTrailingSeparators(string s)
+        {
+            while (s.Length > 0 && SourceFileResolver.IsPathSeparator(s[s.Length - 1]))
+            {
+                s = s.Substring(0, s.Length - 1);
```

Comment:
If I am not mistaken, you are adding the code to PathUtilities in this change too. How does this makes things Ok?